### PR TITLE
update mutation mapper library

### LINKED ISSUE #1276: Mutations table should also show how many samples are mutated
In the example below, the text above the table should be:

Showing 10 mutation(s) in 8 sample(s)

http://bit.ly/1qFC7yF

### PR DESCRIPTION
**Pull request by onursumer**
# What? Why?
Fix #1276, #1379, and #1617.

Changes proposed in this pull request:
See https://github.com/cBioPortal/mutation-mapper/pull/52 for details.

# Checks
- [ ] Runs on Heroku.
- [ ] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Follows the [Google Style Guide](https://github.com/google/styleguide).
- [ ] Make sure your commit messages end with a Signed-off-by string (this line
  can be automatically added by git if you run the `git-commit` command with
  the `-s` option)
- [ ] If this is a feature, the PR is to rc. If this is a bug fix, the PR is to
  hotfix.

# Any screenshots or GIFs?
If this is a new visual feature please add a before/after screenshot or gif
here with e.g. [GifGrabber](http://www.gifgrabber.com/).

# Notify reviewers
If your pull request involves changes to an existing file, one or more of the
people that edited before you might be good candidates to review it. Please use
`git blame <filename>` to determine that and notify them here. Otherwise notify
everybody in the core team:
@cBioPortal/core-developers

### CODE REVIEW

File: portal/src/main/webapp/js/lib/mutationMapper.js [hunk 1]
```diff
@@ -16020,6 +16061,24 @@ function MutationDetailsTable(options, gene, mutationUtil, dataProxies, dataMana
 		}
 	}
 
+	function uniqueSamples(rowData)
+	{
+		var samples = {};
+
+		_.each(rowData, function(data, index) {
+			// assuming only the first element contains the datum
+			var mutation = data[0].mutation;
```
Comment (zhx828) [inline portal/src/main/webapp/js/lib/mutationMapper.js]: You also assume data will always be array?

Comment (onursumer) [inline portal/src/main/webapp/js/lib/mutationMapper.js]: Actually assuming that we are passing an array of row data to this function. And the data is retrieved by this API function: https://datatables.net/reference/api/data()

File: portal/src/main/webapp/WEB-INF/jsp/mutation/mutationMapperTemplates.html [hunk 2]
```diff
@@ -351,69 +351,65 @@
...
 </script>
 
 <script type="text/template" id="mutation_table_cna_template">
-	<label alt='{{cnaTip}}' class='simple-tip {{cnaClass}}'>{{cna}}</label>
+	<span alt='{{cnaTip}}' class='simple-tip {{cnaClass}}'>{{cna}}</span>
 </script>
 
 <script type="text/template" id="mutation_table_cosmic_template">
-	<label class='{{cosmicClass}}'>{{cosmicCount}}</label>
+	<span class='{{cosmicClass}}'>{{cosmicCount}}</span>
```
Comment (pieterlukasse) [inline portal/src/main/webapp/WEB-INF/jsp/mutation/mutationMapperTemplates.html]: do we need to add the `alt` attribute here to keep it 508 compliant? @fedde-s can you comment on this?

**Comment by jjgao** [comment]
@onursumer what's the status of this?

**Comment by onursumer** [comment]
@jjgao : As far as I tested, there is no functional issues. It should be safe to merge.

File: portal/src/main/webapp/WEB-INF/jsp/mutation/mutationMapperTemplates.html (hunk 2 above)
Comment (fedde-s) [inline portal/src/main/webapp/WEB-INF/jsp/mutation/mutationMapperTemplates.html]: @pieterlukasse: Assistive technologies will parse `<label>` tags as labels for input elements/form controls. All such inputs require a label tag for accessibility and Section 508 compliance, but I believe that these spans here are not actually labels for inputs. If that’s correct, removing the `<label>` tags actually improves accessibility.

The `alt=` attribute has no meaning for other elements than `` (it’s just sometimes confused with `title=`), and in this file it’s only used as a custom attribute to be read by the qTip library. That’s no issue for accessibility. 👍

**Comment by fedde-s** [comment]
Running a quick accessibility scan shows only five Section 508 violations (accessibility issues) in the views affected by this PR, but these were not introduced by this PR. They are related to the network tab, such as the textless buttons introduced in commit 153f60bd0598c81a72b969f210d8d4aaf9a92419, which need a textual alternative for the icons, for instance in a Bootstrap `<span class="sr-only">`.

**Pull request merged**